src/Features/Core/Portable/UseConditionalExpression/UseConditionalExpressionForAssignmentHelpers.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.UseConditionalExpression
+{
+    internal static class UseConditionalExpressionForAssignmentHelpers 
+    {
+        public static bool TryMatchPattern(
+            IConditionalOperation ifOperation,
+            out IVariableDeclarationGroupOperation localDeclarationStatement,
+            out ISimpleAssignmentOperation trueAssignmentOperation,
+            out ISimpleAssignmentOperation falseAssignmentOperation)
+        {
+            localDeclarationStatement = null;
+            trueAssignmentOperation = null;
+            falseAssignmentOperation = null;
+
+            var parentBlock = ifOperation.Parent as IBlockOperation;
+            if (parentBlock == null)
+            {
+                return false;
+            }
+
+            // var syntaxFacts = GetSyntaxFactsService();
+            var ifIndex = parentBlock.Operations.IndexOf(ifOperation);
+            if (ifIndex <= 0)
+            {
+                return false;
+            }
+
+            localDeclarationStatement = parentBlock.Operations[ifIndex - 1] as IVariableDeclarationGroupOperation;
+            if (localDeclarationStatement == null)
```

Comment:
I'm really surprised by this... this seems like an arbitrary restriction. Why don't we offer to simplify assignment in these cases?
```c#
int local = ...
// many statements
if (condition)
    local = a;
else
    local = b;
```
to
```c#
int local = ...
// many statements
local = condition ? a : b;
```
and in cases where there's no local declaration at all? with parameters? fields? properties?

Reply:
The fact that *if there happens to be* a declaration right before the if statement, you can later simplify that into one statement, shouldn't disable this code fix in other cases.

Reply:
It isn't intended to.  This was just the code to make sure that we detect the pattern where there is a local-decl right before the if, so we merge it properly.

Reply:
Thanks